pch/main-pch.hpp

```diff
@@ -1,3 +1,15 @@
+// Workaround to a bug in libstdc++ prior to GCC 11 causing
+// Error: attempt to self move assign.
+// when compiling with _GLIBCXX_DEBUG
+// see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=85828
+#if defined(_GLIBCXX_DEBUG) && defined(__GNUC__) && (__GNUC__ < 11)
+#include <debug/macros.h>
+#undef __glibcxx_check_self_move_assign
+#define __glibcxx_check_self_move_assign(x)
+#ifdef __glibcxx_check_self_move_assign // supress unused macro warning
```

Comment:
```suggestion
#ifdef __glibcxx_check_self_move_assign // suppress unused macro warning
```
Yes, I know I'm being compulsive...